lib/personalized_search/matched_product_v2.dart

```diff
@@ -0,0 +1,180 @@
+import 'package:openfoodfacts/model/Attribute.dart';
+import 'package:openfoodfacts/model/AttributeGroup.dart';
+import 'package:openfoodfacts/model/Product.dart';
+import 'package:openfoodfacts/personalized_search/preference_importance.dart';
+import 'package:openfoodfacts/personalized_search/product_preferences_manager.dart';
+
+/// Match and score of a [Product] vs. Preferences, v2
+///
+/// cf. https://github.com/openfoodfacts/smooth-app/issues/1894
+/// cf. https://github.com/openfoodfacts/openfoodfacts-server/blob/main/html/js/product-search.js
+/// cf. https://openfoodfacts.org/js/product-search.js
+
+enum MatchedProductStatusV2 {
+  /// Very good match (very good score)
+  VERY_GOOD,
+
+  /// Good match (good score)
+  GOOD,
+
+  /// Poor match (poor score)
+  POOR,
+
+  /// Unknown match (at least 50% unknown attributes or one mandatory unknown)
+  UNKNOWN,
+
+  /// May not match (at least one mandatory attribute that may not match)
+  MAY,
```

Comment:
Can we keep the full names? VERY_GOOD_MATCH, MAY_NOT_MATCH etc.? I think it would be clearer. And it's easier if all SDKs use the same names.